### PR TITLE
fix(selenium): ensure switchTo() does not fail using SafariDriver

### PR DESCRIPTION
**Pull request by Zidious**
So, this was a strange one. The way we were opening `about:blank` and handles was a bit strange. Using the Safari driver going to any website would would cause `NoSuchWindowException` error. This would consistently throw when testing locally. After some digging, it appears when we called `driver.executeScript("window.open('about:blank', '_blank')");` and attempt to `switchTo()` to the new window it would cause open the `about:blank` page **and** cause the current handle (the page being analyzed) to also navigate to the  `about:blank` page, it appears the window handle would reference itself and not the newly opened page. 

To avoid doing `executeScript` and checking window handles, we can simply lean on Selenium for switching to a new window that is a tab and safely navigate to `about:blank`. 

I also noticed `WebDriverExtensionsTest.java` was missing a tests, so I added some there. 

Closes: https://github.com/dequelabs/axe-core-maven-html/issues/391

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -42,7 +42,9 @@ updates:
     ignore:
       # TODO: https://github.com/dequelabs/axe-core-maven-html/issues/378
       - dependency-name: "org.seleniumhq.selenium:selenium-java"
-        versions: [">=4.14.0"]
+        versions: [ ">=4.14.0" ]
+      - dependency-name: "org.mockito:mockito-core"
+        versions: [ ">=5.0.0" ]
```
Comment (Zidious) [inline .github/dependabot.yml:47]: https://github.com/dequelabs/axe-core-maven-html/issues/393

File: selenium/pom.xml [hunk 2]
```diff
@@ -1,215 +1,224 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (Zidious) [inline selenium/pom.xml:2]: Apologies - this must have been formatting

File: selenium/src/main/java/com/deque/html/axecore/extensions/WebDriverExtensions.java [hunk 3]
```diff
@@ -121,11 +120,7 @@ public static String openAboutBlank(final WebDriver webDriver) {
     String currentWindow = webDriver.getWindowHandle();
 
     try {
-      JavascriptExecutor driver = (JavascriptExecutor) webDriver;
-      driver.executeScript("window.open('about:blank', '_blank')");
-      ArrayList<String> handles = new ArrayList<String>(webDriver.getWindowHandles());
-      String abHandle = handles.get(handles.size() - 1);
-      webDriver.switchTo().window(abHandle);
```
Comment (Zidious) [inline selenium/src/main/java/com/deque/html/axecore/extensions/WebDriverExtensions.java:128]: I tried to look at the git history to see why we did it this way vs `newWindow(..)` but couldn't find anything

File: selenium/pom.xml [hunk 4]
```diff
@@ -1,215 +1,224 @@
...
-  <version>4.7.0</version>
-  <packaging>jar</packaging>
-  <parent>
-    <groupId>com.deque.html.axe-core</groupId>
-    <artifactId>virtual-superpackage</artifactId>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>selenium</artifactId>
     <version>4.7.0</version>
```
Comment (michael-siek) [inline selenium/pom.xml:6]: Should this be 4.8.0?

Comment (Zidious) [reply selenium/pom.xml:6]: https://github.com/dequelabs/axe-core-maven-html/pull/398

**Comment by Zidious** [comment]
Unfortunately, `newWindow` is a Selenium 4 only function. We need to amend the solution to backwards support Selenium 3. 

Putting this into draft until we can introduce a breaking change into our API's

**Pull request closed without merging**